src/test/java/org/jenkinsci/plugins/badge/actions/JobBadgeActionTest.java

```diff
@@ -59,45 +70,84 @@ public static void createAction() throws Exception {
         notBuiltAction = new JobBadgeAction(notBuiltJob);
     }
 
-    @Test
-    public void testGetIconFileName() {
-        assertThat(notBuiltAction.getIconFileName(), is(nullValue()));
-        assertThat(successfulAction.getIconFileName(), is(nullValue()));
+    @Before
+    public void createWebClient() {
+        webClient = j.createWebClient();
     }
 
-    @Test
-    public void testGetIconClassName() {
-        assertThat(notBuiltAction.getIconClassName(), is("symbol-shield-outline plugin-ionicons-api"));
-        assertThat(successfulAction.getIconClassName(), is("symbol-shield-outline plugin-ionicons-api"));
-    }
+    @Before
+    public void setUpActionForBadgeToBuildAndRun() throws IOException {
+        // Create an instance of JobBadgeAction
+        jobBadgeActionBuildAndRun = new JobBadgeAction(successfulAction.project);
 
-    @Test
-    public void testGetDisplayName() {
-        assertThat(notBuiltAction.getDisplayName(), is("Embeddable Build Status"));
-        assertThat(successfulAction.getDisplayName(), is("Embeddable Build Status"));
-    }
+        // Get the Jenkins URL
+        jenkinsUrl = j.getURL().toString() + "job/" + jobBadgeActionBuildAndRun.getUrlEncodedFullName() + "/" + "badge/icon";
+        badgeUrl = jenkinsUrl + "?build=1&style=style&subject=subject&status=status&color=green&config=config&animatedOverlayColor=animatedOverlayColor&link=link";
+        // Open a connection to the badge URL
+        URL url = new URL(badgeUrl);
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+        connection.setRequestMethod("GET");
```

Comment:
I don't understand how this connection is used.  As far as I can tell from the [javadoc](https://javadoc.scijava.org/Java11/java.base/java/net/HttpURLConnection.html), this must be unused